middlewares/devFlag.ts

```diff
@@ -0,0 +1,14 @@
+import { NextFunction, Request, Response } from "express";
+import { CustomRequest, CustomResponse } from "../types/global";
+
+export const devFlagMiddleware = (req: CustomRequest, res: CustomResponse, next: NextFunction) => {
```

Comment:
1. Dependency on boom: The code assumes you have a library named boom installed in your project with a notFound function that returns a 404 error.  If boom is not available, the catch block will be necessary to handle potential errors.

2. Error Handling:  While the current catch block passes the error (err) to the next function, it doesn't provide any specific handling within the middleware itself. You might want to consider logging the error or returning a more informative response depending on your application's needs.

3. Alternative for boom:  If you don't want to rely on boom, you can achieve the same functionality using vanilla Node.js functionalities:

Reply:
1. Yes we are using **express-boom** to handle errors in the whole repo, this is why I  have used res.boom. I didn't use res.status just to maintain the consistency in the repo.
2. I have added a logger for handling errors with more information.
3.  Yes sir, we can either use ```res.status(404).send("Route not found")``` and many more libraries for this. But as in the whole repo we are using boom this is why I am also using this.